### ISSUE TITLE
[JP server] Fails on factory in infrastructure

### ISSUE THREAD

**Issue by Mion354D** [MaaAssistantArknights/MaaAssistantArknights#3413]
### Before raising the issue...

- [X] I understand that Issues are for feedback and problem solving, not trolling the comments section, and will provide as much information as possible to help solve the problem.
- [X] I filled in a short, clear title so that developers could quickly identify the general problem when going through the issue list. Instead of "Some suggestions", "Stuck", etc.
- [X] I have checked the latest beta update and there is no mention of the bug being fixed.

### Description

MAA assigns only one operator not three in the first factory and ignores other factories. Works fine on other parts of the infrastructure.

MAA works under Alas.

### Log files

[maa.log](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/10443740/maa.log)

### Screenshots

![Snipaste_2023-01-18_16-11-14](https://user-images.githubusercontent.com/7450570/213121210-901f4113-4167-4c75-b37f-6b87cf46e037.jpg)
![Snipaste_2023-01-18_16-26-38](https://user-images.githubusercontent.com/7450570/213121216-9db42638-93b9-4909-949c-dabe784280c6.jpg)
![Snipaste_2023-01-18_16-27-04](https://user-images.githubusercontent.com/7450570/213121219-706b4434-7b10-4c0e-a9fb-829432a196f6.jpg)

**Comment by MistEO** [comment]
Do you have MAA's own log? instead of ALAS' log. It's `<your MAA dir>/debug/asst.log`

**Comment by Mion354D** [comment]
> Do you have MAA's own log? instead of ALAS' log. It's `<your MAA dir>/debug/asst.log`

[asst.log](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/10450235/asst.log)

**Comment by MistEO** [comment]
```log
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_exp2, ] mood 0.152941 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd2, ] mood 0.152941 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd1, ] mood 0.129412 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_gold1, ] mood 0.0705882 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd2, ] mood 0.0705882 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_exp2, ] mood 0.141176 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd_add1, ] mood 0.141176 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_limit&cost1, ] mood 0.0823529 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd2, ] mood 0.0705882 threshold 0.3
[2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_limit&cost1, bskill_man_exp1, ] mood 0.0588235 threshold 0.3
```

Most of your opers are out of mood, so they are not selected. If this is not the case, plz help record a video so we can see exactly what happened.

**Comment by Mion354D** [comment]
Mfg > Trade > Power > Control > Reception > Office > 

> ```
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_exp2, ] mood 0.152941 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd2, ] mood 0.152941 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd1, ] mood 0.129412 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_gold1, ] mood 0.0705882 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd2, ] mood 0.0705882 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_exp2, ] mood 0.141176 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd_add1, ] mood 0.141176 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_limit&cost1, ] mood 0.0823529 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_spd2, ] mood 0.0705882 threshold 0.3
> [2023-01-19 03:50:30.726][TRC][Px7b74][Txd04] [bskill_man_limit&cost1, bskill_man_exp1, ] mood 0.0588235 threshold 0.3
> ```
> 
> Most of your operators are out of mood, so they are not selected. If this is not the case, plz help record a video so we can see exactly what happened.

You are right, I found that the reason for low mood is taht Maa did not assign the operators into dorms. 
I only enabled the dorm and started maa, the log attached below.

![image](https://user-images.githubusercontent.com/7450570/214079108-67296bd0-ac64-46a5-a7e7-6e7e01c159a2.png)

[asst.log](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/10481139/asst.log)

**Comment by Mion354D** [comment]
And I recorded a video if it helps.
https://1drv.ms/v/s!ArHxMhRCxYxCgt1pGHYoMVmU0RO-cA?e=wOl2Cs

**Comment by Mion354D** [comment]
Seems fixed by [f33e004e3c918c7e3501328449cc92cf8a5c434a](https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/3602)
